pulsar-broker/src/test/java/org/apache/pulsar/broker/intercept/CounterBrokerInterceptor.java

```diff
@@ -280,7 +294,13 @@ public void txnEnded(String txnID, long txnAction) {
 
     @Override
     public void initialize(PulsarService pulsarService) throws Exception {
+        var meter = pulsarService.getOpenTelemetry().getMeter();
 
+        messageCounter = meter
+                .counterBuilder(MESSAGE_COUNTER)
+                .setUnit("{message}")
+                .setDescription("The total number of messages produced.")
```

Comment:
Not sure I get the full intent here, as this metric is in a test.

If you want to learn how to add a metric it's best by doing it for a real metric.

Reply:
This just show how to customize a metric in an `BrokerInterceptor`.

In fact, this will be a business-customized metric, and the `MESSAGE_COUNTER` here is just a demo.

The background comes from this https://github.com/apache/pulsar/issues/20772

Reply:
Got it. Thanks.
So if I understand correctly you would like to have the ability to add your own metrics when you write your own `BrokerInterceptor`.

So this is not a real PR, but a Draft PR - I would convert it to such, so it won't by mistake gets merged.

The same use case you described exists for any "plugin" people can add to the broker: Function, Offloader plugin, Authentication plugin and many more.

As I wrote in the parent PIP of OTel (pip-264), each such plugin should create their own meter, which reflect their own namespace.

Reply:
> So this is not a real PR, but a Draft PR - I would convert it to such, so it won't by mistake gets merged.

Yes, you are right.

I think this PR is also necessary; this PR can verify that using `pulsarService.getOpenTelemetry().getMeter()` in an interceptor to expose metrics is as expected.